### PR TITLE
WIP/Enums: Support rendering enum fields in timeseries

### PR DESCRIPTION
**Pull request by ryantxu**
This adds a new field type `enum` that maps int<>string properties efficiently.

This was recently added to the SDK, but needs work in the UI to make it configurable

### CODE REVIEW

File: packages/grafana-ui/src/types/icon.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export function getFieldTypeIcon(field?: Field): IconName {
         return 'toggle-on';
       case FieldType.trace:
         return 'info-circle';
+      case FieldType.enum:
+        return 'list-ul';
```
Comment (leeoniya) [inline packages/grafana-ui/src/types/icon.ts:30]: `list-ol`?

Comment (ryantxu) [reply packages/grafana-ui/src/types/icon.ts:30]: 🤔 
image

File: public/app/plugins/panel/timeseries/utils.ts [hunk 2]
```diff
@@ -17,6 +17,41 @@ import { applyNullInsertThreshold } from '@grafana/ui/src/components/GraphNG/nul
...
 
+// mutates fields!
+function unifyEnumFields(frames: DataFrame[]) {
+  let allTexts: string[] = [];
+
+  let frames2: DataFrame[] = frames.map((frame) => {
+    return {
+      ...frame,
+      fields: frame.fields.map((field) => {
+        if (field.type === FieldType.enum) {
```
Comment (ryantxu) [inline public/app/plugins/panel/timeseries/utils.ts:28]: maybe we should check that any enum field exists first?  then we can avoid all the copies for the 99% chance they do not 🤔

Comment (leeoniya) [reply public/app/plugins/panel/timeseries/utils.ts:28]: i dont think there's a lot of perf to gain here. it's basically just `frame.fields.map((field) => field)` for what, like < 30 items?

Comment (ryantxu) [reply public/app/plugins/panel/timeseries/utils.ts:28]: maybe... but I am actually talking about the step above it that will be: `frame => { ...frame }` for as many series as prometheus returns 🤷 30?

Comment (leeoniya) [reply public/app/plugins/panel/timeseries/utils.ts:28]: i'm not fundamentally opposed to adding a precheck :)

**Comment by leeoniya** [comment]
superseded by https://github.com/grafana/grafana/pull/64179

**Pull request closed without merging**